### PR TITLE
DashboardScene: Update timerange in datalinks on change

### LINKED ISSUE #91884: Dashboard Scene: Dashboard link is not preserving timerange
Configuring a dashboard link with the option `include current time range` is not working when the time range changes. 

https://github.com/user-attachments/assets/f6ef0019-25ce-4612-ac8b-7f446a862539

### PR DESCRIPTION
**Pull request by mdvictor**
**What is this feature?**

Fixes a bug where the datalink URL would not refresh the timerange on change if the ranges was included in the URL.

**Why do we need this feature?**

To address the issue in scenes dashboards datalinks

**Who is this feature for?**

Everyone

**Which issue(s) does this PR fix?**:

Fixes #91884

**Special notes for your reviewer:**

Please check that:
- [x] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/contribute/release-notes/#how-to-determine-if-content-belongs-in-whats-new), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/contribute/release-notes/) doc.

### CODE REVIEW

File: public/app/features/dashboard-scene/scene/DashboardLinksControls.tsx [hunk 1]
```diff
@@ -12,18 +15,39 @@ import { LINK_ICON_MAP } from '../settings/links/utils';
...
 
-export function DashboardLinksControls({ links, uid }: Props) {
+interface LinkInfo {
+  href: string;
+  title: string;
+  tooltip: string;
+}
+
+export function DashboardLinksControls({ links, uid, timeRange }: Props) {
+  const timeRangeState = timeRange?.useState();
```
Comment (kaydelaney) [inline public/app/features/dashboard-scene/scene/DashboardLinksControls.tsx:29]: Should timeRange be optional? Not sure if this scenario could happen, but if a timeRange was passed, and then later was undefined, I think React would complain about inconsistent hook usage

**Review by kaydelaney — APPROVED** [review]
Looks good! Just had one question in another comment

File: public/app/features/dashboard-scene/scene/DashboardLinksControls.tsx (hunk 1 above)
Comment (torkelo) [reply public/app/features/dashboard-scene/scene/DashboardLinksControls.tsx:29]: yes, it should. for this component at least

Comment (mdvictor) [reply public/app/features/dashboard-scene/scene/DashboardLinksControls.tsx]: Fixed, thank you!

**Comment by mdvictor** [comment]
I changed some things around here, I think this might be better. The old code had issues with variables and wouldn't update the links in the effect when variables changed. So instead I reckon we can just timerange.useState in the controls to trigger the state change effect and rerender the controls when the timerange updates, similar to how variables trigger a rerender of the controls when those change. This way the url is synced properly.

**Pull request merged**